README.md

```diff
@@ -1 +1,118 @@
+
+However, it might come to the point where we need to actually do raw DOM manipulation on the directive itself. One problem - the controller is initiated *before* the directive is in the DOM. What do we do?
+
+
+### `link`
+
+This is where the `link` function comes in. If we specify this, we get a function executed *after* the directive has been mounted in the DOM. We also then get the DOM element passed through to us as the second parameter.
+
+```js
+function someDirective() {
+  return {
+    template: [
+	  '<div>',
+	    'Hello!',
+	  '</div>'
+	].join(''),
+    link: function ($scope, $element, $attrs) {
+
+    }
+  };
+}
+
+angular
+  .module('app', [])
+  .directive('someDirective', someDirective);
+```
+
+You can see above that we get three parameters (they must be in that order!). `$scope` - much like what we can inject into our controllers. `$element`, our DOM mounted element, and `$attrs`, an object of all the attributes used when the directive is initiated.
+
+### `compile`
+
+Compile allows us to manipulate the element before it is inserted into the DOM. This runs in between the `controller` and the `link` function. This gives us access to the unmounted DOM node, and the attributes.
+
+```js
+function someDirective() {
+  return {
+    template: [
+      '<div>',
+        'Hello!',
+      '</div>'
+    ].join(''),
+    compile: function ($element, $attrs) {
+
+    }
+  };
+}
+
+angular
+  .module('app', [])
+  .directive('someDirective', someDirective);
+```
+
+In our compile function, we'd then return an object with `pre` and `post` values. Both of these are functions.
+
+These are `pre-link` and `post-link` functions, giving us access to every single point in the directive lifecycle.
+
+Take a look at this diagram of the lifecycle:
+
+![http://i.stack.imgur.com/XRDs6.png](Image)
+
+This means that if we put some `console.log` calls in our compile function, as so:
+
+```js
+function someDirective() {
+  return {
+    template: [
+      '<div>',
+        'Hello!',
+      '</div>'
+    ].join(''),
+    controller: function () {
+        console.log('controller');
+    },
+    compile: function ($element, $attrs) {
+        console.log('compile');
+
+        return {
+            pre: function (scope, element, attrs) {
+                console.log('pre-link');
+            },
+            post: function (scope, element, attrs) {
+                console.log('post-link');
+            }
+        }
+    }
+  };
+}
+
+angular
+  .module('app', [])
+  .directive('someDirective', someDirective);
+```
+
+We would get this printed out in the console:
+
+```js
+compile
+controller
+pre-link
+post-link
+```
+
+This gives us full control over what we want to do with our directives.
```

Comment:
Can you add some context to the conclusion? Do you find that you need compile functions more than link, or vis versa? What are the use cases for these attributes?